File: Examples/Node.js/Document/MergeSelectedSlides.js
// ExStart:1
var fs = require('fs');
var assert = require('assert');
var SlidesApi = require('asposeslidescloud');
var StorageApi = require('asposestoragecloud');
var configProps = require('../Data/config.json');
var data_path = '../Data/';

var AppSID = configProps.app_sid;
var AppKey = configProps.api_key;
var outFolder = configProps.out_folder;
var config = {'appSid':AppSID,'apiKey':AppKey , 'debug' : true};

// Instantiate Aspose Storage API SDK
var storageApi = new StorageApi(config);
// Instantiate Aspose.Slides API SDK
var slidesApi = new SlidesApi(config);

// Set input file name
var name = "sample.pptx";
var mergeFile1 = "welcome.pptx";
var mergeFile2 = "demo.pptx";

var presentationToMerge1 = {
		'Path' : mergeFile1,
		'Slides' : [2]
};

var presentationToMerge2 = {
		'Path' : mergeFile2,
		'Slides' : [1]
};

var orderedMergeRequestBody = {
		'Presentations' : [presentationToMerge1, presentationToMerge2]
};

try {

// Upload files to aspose cloud storage
storageApi.PutCreate(name, null, null, data_path + name , function(responseMessage) {
	assert.equal(responseMessage.status, 'OK');

	storageApi.PutCreate(mergeFile1, null, null, data_path + mergeFile1 , function(responseMessage) {
		assert.equal(responseMessage.status, 'OK');

		storageApi.PutCreate(mergeFile2, null, null, data_path + mergeFile2 , function(responseMessage) {
			assert.equal(responseMessage.status, 'OK');

			// Invoke Aspose.Slides Cloud SDK API to take selected slides from multiple presentation files and then merge
			slidesApi.PutPresentationMerge(name, null, null, orderedMergeRequestBody, function(responseMessage) {
					assert.equal(responseMessage.status, 'OK');

					// Download presentation from cloud storage
					storageApi.GetDownload(name, null, null, function(responseMessage) {
						assert.equal(responseMessage.status, 'OK');
						var writeStream = fs.createWriteStream(data_path + name);
						writeStream.write(responseMessage.body);
						});
				});
			});
		});
	});

}catch (e) {
  console.log("exception in example");
  console.log(e);
}
//ExEnd:1

File: Examples/Python/MergeSelectedSlides.py
import asposeslidescloud
from asposeslidescloud.SlidesApi import SlidesApi
from asposeslidescloud.SlidesApi import ApiException
from asposeslidescloud.models import OrderedMergeRequest
from asposeslidescloud.models import PresentationToMerge

import asposestoragecloud
from asposestoragecloud.StorageApi import StorageApi
from asposestoragecloud.StorageApi import ResponseMessage

apiKey = "XXXXX" #sepcify App Key
appSid = "XXXXX" #sepcify App SID
apiServer = "http://api.aspose.com/v1.1"
data_folder = "../../data/"

#Instantiate Aspose Storage API SDK
storage_apiClient = asposestoragecloud.ApiClient.ApiClient(apiKey, appSid, True)
storageApi = StorageApi(storage_apiClient)
#Instantiate Aspose Slides API SDK
api_client = asposeslidescloud.ApiClient.ApiClient(apiKey, appSid, True)
slidesApi = SlidesApi(api_client);

#set input file name
name =  "sample.pptx"
mergeFile1 = "welcome.pptx"
mergeFile2 = "demo.pptx"

presentationToMerge1 = PresentationToMerge.PresentationToMerge()
presentationToMerge1.Path = mergeFile1
presentationToMerge1.Slides = [2]

presentationToMerge2 = PresentationToMerge.PresentationToMerge()
presentationToMerge2.Path = mergeFile2
presentationToMerge2.Slides = [1]


body = OrderedMergeRequest.OrderedMergeRequest()
body.Presentations = [presentationToMerge1, presentationToMerge2]


try:
    #upload file to aspose cloud storage
    response = storageApi.PutCreate(name, data_folder + name)
    response = storageApi.PutCreate(mergeFile1, data_folder + mergeFile1)
    response = storageApi.PutCreate(mergeFile2, data_folder + mergeFile2)

    #invoke Aspose.Slides Cloud SDK API to take selected slides from multiple presentation files and then merge
    response = slidesApi.PutPresentationMerge(name, body)

    if response.Status == "OK":
        #download presentation from cloud storage
        response = storageApi.GetDownload(Path=name)
        outfilename = "c:/temp/" + name
        with open(outfilename, 'wb') as f:
                    for chunk in response.InputStream:
                        f.write(chunk)

except ApiException as ex:
            print "ApiException:"
            print "Code:" + str(ex.code)
            print "Message:" + ex.message
